### ISSUE TITLE
Razor SDK produces two assemblies for a Razor Class Library project

### ISSUE THREAD

**Issue by evil-shrike** [aspnet/Razor#2396]
#1809

Hi.
I'm trying to migrate my solution from 2.0 (see https://github.com/aspnet/Razor/issues/1809#issue-281111990) to the new 2.1 Razon SDK. 
I have moved all my views ("Views" folder) from a project into a new RazorSDK-project "Views". It references the main back (to have Model classes) and package Microsoft.AspNetCore.Mvc, that's all.
Compiling these two projects produces two "Views" assemblies! 

Here's the most simpified demo. 
* Create a new empty Web .net core project
* Create a new empty Razor Class Library project
* Reference the first project from razor project
* change razor' TargetFramework  from netstandard to netcoreapp2.1 (why do you generate it as netstandard btw? it makes it fail to compile in almost all cases as referenced web-project won't be netstandard).

Here're two csproj:

### RazorSdkTest.Views.csproj:
```
<Project Sdk="Microsoft.NET.Sdk.Razor">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.1.0" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\RazorSdkTest\RazorSdkTest.csproj" />
  </ItemGroup>
</Project>
```

### RazorSdkTest.csproj:
```
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Folder Include="wwwroot\" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.App" />
  </ItemGroup>

</Project>
```

Now build and get the following assemblies in RazorSdkTest.Views\bin\Debug\netcoreapp2.1: 
```
RazorSdkTest.Views.dll
RazorSdkTest.Views.pdb
RazorSdkTest.Views.deps.json
RazorSdkTest.Views.Views.dll
RazorSdkTest.Views.Views.pdb
```

RazorSdkTest.Views.dll is empty, while RazorSdkTest.Views.Views.dll contains all views.

**Comment by manigandham** [comment]
What is the problem? 

Views are compiled into a separate assembly file by the [Razor SDK](https://docs.microsoft.com/en-us/aspnet/core/mvc/razor-pages/sdk?view=aspnetcore-2.1) with the name `YOURPROJECT.Views.dll`  which is what you're seeing. `YOURPROJECT.dll` will contain the rest of your app logic, but since you don't have anything in that project than that assembly will be empty.

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @evil-shrike.
@pranavkm, can you please look into this? Thanks!

**Comment by evil-shrike** [comment]
The problem is that it's a bit unclear from the doc how to use view precompilation. 
I had `<MvcRazorCompileOnPublish>true</MvcRazorCompileOnPublish>` in my Web-project and when I changed Sdk from "Microsoft.NET.Sdk.Web" to "Microsoft.NET.Sdk.Razor" nothing changed - rebuilding didn't produce a new assembly. So I decided that I should add a new project for Views with Razor SDK. And for that project I got two assemblies. I looked weird.
Now I realized that it was enough just to remove MvcRazorCompileOnPublish and set SDK to "Microsoft.NET.Sdk.Razor".
Thanks.
